Examples/ExampleView.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  ExampleView.swift
+//  CollectionKitExample
+//
+//  Created by Luke Zhao on 2017-09-04.
+//  Copyright © 2017 lkzhao. All rights reserved.
+//
+
+import UIKit
+import CollectionKit
+
+class ExampleView: UIView {
+  let titleLabel: UILabel = {
+    let titleLabel = UILabel()
+    titleLabel.numberOfLines = 0
+    titleLabel.font = .boldSystemFont(ofSize: 20)
+    return titleLabel
+  }()
+
+  let cardView: UIView = {
+    let cardView = UIView()
+    cardView.backgroundColor = .white
+    cardView.layer.cornerRadius = 8
+    cardView.layer.shadowColor = UIColor.black.cgColor
+    cardView.layer.shadowOffset = CGSize(width: 0, height: 12)
+    cardView.layer.shadowRadius = 10
+    cardView.layer.shadowOpacity = 0.1
+    cardView.layer.borderColor = UIColor(white: 0, alpha: 0.1).cgColor
+    cardView.layer.borderWidth = 0.5
+    return cardView
+  }()
+
+  private var contentVC: UIViewController?
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    addSubview(titleLabel)
+    addSubview(cardView)
+  }
+
+  public required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+    let size = titleLabel.sizeThatFits(bounds.insetBy(dx: 16, dy: 10).size)
+    titleLabel.frame = CGRect(origin: CGPoint(x: 0, y: 10), size: size)
+    let labelHeight = titleLabel.frame.maxY + 10
+    cardView.frame = CGRect(x: 0, y: labelHeight, width: bounds.width, height: bounds.height - labelHeight)
+    contentVC?.view.frame = cardView.bounds
+  }
+
+  func populate(title: String,
+                contentViewControllerType: UIViewController.Type) {
+    titleLabel.text = title
+    contentVC?.view.removeFromSuperview()
```

Comment:
How about have a file that can generate a collectionProvider, for this card it contains a title and collectionView.

```swift
ArticleProvider() -> AnyCollectionViewProvider {
  // Configure a ArticleExampleProvider and return.
}

let examples: [(String, UIViewController.Type)] = [
     ("Article example", ArticleProvider),
     :
     :
   ]
```

When populate this card, we only set collectionView's provider? It will show the ability to reuse a provider.

Here view controller is more like a wrapper, for `ArticleExampleViewController`, it is quite easy to extract a provider, but for `messagesViewController`, it seems require a lot of work.

I think for now, this is fine.

Reply:
I think that adds a lot more complexity to the demo app. Currently each example is well decoupled and we demonstrated how we setup each collectionView accordingly. We can add another example that demonstrate Reusable provider class.

Reply:
Ok, every thing is quite easy to add and remove!